### PR TITLE
Qa krishan

### PR DESCRIPTION
**Pull request by gem-krishanpandey**
@gem-rishabhrastogi please review the code.

### CODE REVIEW

File: src/main/java/com/qa/hris/commonUtils/HrisUtils.java [hunk 1]
```diff
@@ -28,8 +22,8 @@ public class HrisUtils extends DriverAction {
     public void signIn() {
         try {
             waitSec(5);
-            if (isExist(Login_Locators.loginButton)) {
-                click(Login_Locators.loginButton, "Click on Login Button", "Clicked Login Button");
+            if (isExist(LoginLocators.loginButton)) {
+                click(LoginLocators.loginButton, "Click on Login Button", "Clicked Login Button");
             }
         } catch (Exception exception) {
             GemTestReporter.addTestStep("Check visibility of Login Button", "Login button is not present", STATUS.FAIL, takeSnapShot());
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/commonUtils/HrisUtils.java:29]: Not a good idea to throw an exception inside the catch block.

File: src/main/java/com/qa/hris/commonUtils/HrisUtils.java [hunk 2]
```diff
@@ -46,16 +40,16 @@ public void enterCredentials(String credentialType) throws IOException {
...
                 case "username":
-                    typeText(Login_Locators.credentials(credentialType),"Enter the username","Username entered successfully", readProperties(credentialType));
+                    typeText(LoginLocators.credentials(credentialType), "Enter the username", "Username entered successfully", readProperties(credentialType));
                     break;
                 case "password":
-                    typeText(Login_Locators.credentials(credentialType),"Enter the password","Password entered successfully", readProperties(credentialType));
+                    typeText(LoginLocators.credentials(credentialType), "Enter the password", "Password entered successfully", readProperties(credentialType));
                     break;
                 default:
                     System.out.println("Please enter a valid Input");
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/commonUtils/HrisUtils.java:49]: Please refrain from using S.O.P statements. If you want some message to be printed out to user, it must be added as a step in your report.

File: src/main/java/com/qa/hris/commonUtils/HrisUtils.java [hunk 3]
```diff
@@ -67,7 +61,7 @@ public void enterCredentials(String credentialType) throws IOException {
     public void login() {
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/commonUtils/HrisUtils.java:61]: Not a good idea to create two functions with similar meanings (login() and signIn())

File: src/main/java/com/qa/hris/commonUtils/HrisUtils.java [hunk 4]
```diff
@@ -87,7 +81,7 @@ public void verifyLogin() {
         try {
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/commonUtils/HrisUtils.java:81]: In readProperties() - The properties file path should not be hard-coded here.

File: src/main/java/com/qa/hris/commonUtils/TrainingUtils.java [hunk 5]
```diff
@@ -2,26 +2,25 @@
...
-        if (isExist(locator)){
-            typeText(locator,"Verify user enters the data in "+nameOfField,"Data is entered successfully",dataToBeEntered);
-        }}
-        catch (Exception exception){
-            GemTestReporter.addTestStep("Verify user enters the data in "+nameOfField, "Exception arises", STATUS.ERR, takeSnapShot());
+    public static void enterTextInField(By locator, String dataToBeEntered, String nameOfField) {
+        try {
+            if (isExist(locator)) {
+                typeText(locator, "Verify user enters the data in " + nameOfField, "Data is entered successfully", dataToBeEntered);
+            }
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/commonUtils/TrainingUtils.java:16]: Add the else case to cover the scenario when field is somehow not present on screen.

File: src/main/java/com/qa/hris/locators/LoginLocators.java [hunk 6]
```diff
@@ -2,10 +2,14 @@
...
 import org.openqa.selenium.By;
 
-public class Login_Locators {
+public class LoginLocators {
 
     public static By loginButton = By.xpath("//button[@class='login-btn']");
 
+    public static By nextButton = By.id("idSIButton9");
+
+    public static By image = By.xpath("//img[@class='img-pos']");
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/locators/LoginLocators.java:11]: Assign meaningful names to variables

**Review by gem-rishabhrastogi — CHANGES_REQUESTED** [review]
Added some suggestions. Please work on those.

**Review by gem-rishabhrastogi — APPROVED** [review]
Good to go.

**Comment by gem-krishanpandey** [comment]
Hi @gem-rishabhrastogi kindly review the code.

File: src/main/java/com/qa/hris/stepDefinition/Training.java [hunk 7]
```diff
@@ -170,19 +169,19 @@ public void userAddDateToOfferDate() {
...
-    @Then("User enters his phone Number")
-    public void userEntersHisPhoneNumber() {
-        TrainingUtils.enterTextInField(TrainingLocators.getFieldData("phoneNumber"), "9878997780", "Phone Number");
+    @Then("User enters his {string} phone Number")
+    public void userEntersHisPhoneNumber(String number) {
+        TrainingUtils.enterTextInField(TrainingLocators.getFieldData("phoneNumber"), number, "Phone Number");
     }
 
     @Then("User selects their experience")
     public void userSelectsTheirExperience() {
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/hris/stepDefinition/Training.java:178]: Take experience as an argument from the feature file

**Review by gem-rishabhrastogi — CHANGES_REQUESTED** [review]
Added a couple of suggestions. Please work on those.

**Review by gem-rishabhrastogi — APPROVED** [review]
Good to go.

**Pull request merged**